formatters/simplecov/simplecov.go

```diff
@@ -41,30 +43,71 @@ func (r Formatter) Format() (formatters.Report, error) {
 		return rep, errors.WithStack(errors.Errorf("could not open coverage file %s", r.Path))
 	}
 
-	m := map[string]input{}
-	err = json.NewDecoder(jf).Decode(&m)
-	if err != nil {
-		return rep, errors.WithStack(err)
-	}
+	var m simplecovJsonFormatterReport
+	decoder := json.NewDecoder(jf)
+	decoder.DisallowUnknownFields()
 
-	gitHead, _ := env.GetHead()
-	for _, v := range m {
-		for n, ls := range v.Coverage {
+	err = decoder.Decode(&m)
+
+	if err != nil {
+		return legacyFormat(r, rep)
+	} else {
+		logrus.Debugf("Analyzing simplecov json output from latest format %s", r.Path)
```

Comment:
nit: would it be possible to move this logic into its own file similar to `legacyFormat` to keep the same structure?

Reply:
I liked the suggestion thanks, addressed. Just couldn't find a more clever name to the new formatter than `jsonFormatter` although it makes sense since it designed to read ouput from the the SimpleCov's JsonFormatter.